segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -150,6 +156,14 @@ public void setupAppend(SetupAppend setupAppend) {
         String newSegment = setupAppend.getSegment();
         UUID writer = setupAppend.getWriterId();
         log.info("Setting up appends for writer: {} on segment: {}", writer, newSegment);
+        if (this.tokenVerifier != null && !tokenVerifier.verifyToken(newSegment,
+                setupAppend.getDelegationToken(), PravegaAuthHandler.PravegaAccessControlEnum.READ)) {
```

Comment:
Is this supposed to be READ? It's about to write.

Reply:
Fixed. Thanks.